src/ucs/config/parser.c

```diff
@@ -1112,12 +1309,27 @@ static void ucs_config_print_doc_line_by_line(const ucs_config_field_t *field,
 }
 
 static ucs_status_t
-ucs_config_parser_parse_field(ucs_config_field_t *field, const char *value, void *var)
+ucs_config_parser_parse_field(const ucs_config_field_t *field,
+                              const char *value, void *var)
 {
-    char syntax_buf[256];
+    char value_buf[512]  = "";
+    char syntax_buf[256] = "";
     int ret;
 
-    ret = field->parser.read(value, var, field->parser.arg);
+    if (ucs_config_is_key_value_field(field)) {
```

Comment:
should not have a special case here, it should be part of ucs_config_sscanf_key_value

Reply:
Yes, I would also prefer it that way.
But with current design we cannot access `field->dfl_value` from the parser.
The easiest way would be to add one more argument to parser.read signature, so that it accepts also default value:
```
    int                      (*read) (const char *buf, void *dest, const void * dfl_value, const void *arg);

```
Would it be ok?

Reply:
As the comment below explains, this is needed to support *partial setting*.
Supposedly we want to allow users to specify only subset of all the supported keys.
For instance, in our TEMP example with allowed keys [front,rear] I can make this setting:
`TEMP=rear:15`  (what is set for `front`?)
This is what I called a *partial setting*, meaning that my config does NOT provide explicit values for all the allowed keys.

Should we allow that? I thought *yes*, but I don't have a strong opinion on that. What's your take on that?
If it's not what we want to support, I just remove this code.

If we allow that, then we need to initialize those remaining keys somehow. And inside parser.read I have no access to default value.
So the idea was to concatenate the default value, followed by current setting. This way we guarantee that a) all fields are initialized, b) current setting is applied on top of default, so it has precedence.

Reply:
i think we should not allow that.
when setting a new config value, the default value in the table becomes irrelevant (this is how all configs work today)

Reply:
Ok, removing